warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryPruningVisitor.java

```diff
@@ -233,6 +233,9 @@ public Object visit(ASTOrNode node, Object data) {
                 return TruthState.TRUE;
             } else if (state == TruthState.FALSE) {
                 // drop this node from the OR it can never be satisfied
+                // if the child was a junction that was pruned, then the child object is a ref node but the child
+                // at index i is not. reassign for consistency.
+                child = node.jjtGetChild(i);
```

Comment:
I'm just starting to look through this, but this caught my eye.  I ran all of the tests in QueryPruningVisitor with a conditional breakpoint here: `chiled !- node.jjtGetChild(i)`

None of the tests hit on this breakpoint.  Is this really needed?

Reply:
I take that back.  I hit on it - still reviewing.